fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/common/StoreTimer.java

```diff
@@ -61,6 +61,8 @@
     protected final Map<Event, Counter> counters;
     @Nonnull
     protected final Map<Event, Counter> timeoutCounters;
+    @Nonnull
+    protected long lastReset;
```

Comment:
Minor nit: You could reset a timer and attempt to diff it within the same millisecond.  Maybe a cleaner idea would be that each `StoreTimer` has a UUID, like:

    @Nonnull
    protected UUID uuid = UUID.randomUUID();

this `UUID` could change each time the time is reset, and the snapshot could carry around the `UUID` from which it was derived.

Reply:
The time units used here and with the timer in general are nanoseconds.  So if you created a snapshot and subsequently reset it in the same millisecond it *seems* you would see that the reset happened after the snapshot was created but perhaps there is a hole here that I am not seeing. Also if we wanted to preserve the possibility that at some point in the future we might want the capability to reset the timer to a particular point in the past vs. wipe it out. then the snapshot could still be good if it wasn't reset past the point in time the snapshot was created.